lib/openstack_taster.rb

```diff
@@ -52,8 +53,21 @@ def initialize(
       .select { |network| network.name == INSTANCE_NETWORK_NAME }.first
   end
 
-  def taste(image_name, settings) # FIXME: Reduce Percieved and Cyclomatic complexity
-    image = @compute_service.images # FIXME: Images over compute service is deprecated
+  # Taste a specified image
+  # @param image_name [String] The name on OpenStack of the image to be tested.
+  # @param settings [Hash] A hash of settings to enable and disable tests, snapshot creation upon failure.
+  # @return [Boolean] success or failure of tests on image.
+  # @note The testing section could be further streamlined by:  
+  #   creating a naming standard for test functionions (i.e. taste_<name>)  
+  #   limiting the parameters of each test to be: instance, distro_username  
+  #   Adding a 'suites' subhash to the settings hash  
+  #   Then that subhash can be iterated over, use eval to call each function,
+  #   appending the suite name to 'taste_' for the function name
+  #   and passing the standardized parameters.
```

Comment:
These ideas are already done, right ?  We probably just need to document it as a standard in the CONTRIBUTING doc.